staging/src/k8s.io/apiserver/pkg/endpoints/filters/audit.go

```diff
@@ -91,8 +60,8 @@ var _ http.Hijacker = &fancyResponseWriterDelegator{}
 // 2. the response line containing:
 //    - the unique id from 1
 //    - response code
-func WithAudit(handler http.Handler, requestContextMapper request.RequestContextMapper, out io.Writer) http.Handler {
-	if out == nil {
+func WithAudit(handler http.Handler, requestContextMapper request.RequestContextMapper, sink audit.Sink, policy *auditinternal.Policy, longRunningCheck filters.LongRunningRequestCheck) http.Handler {
```

Comment:
Actually, we have put all filters together in the big refactoring earlier this year. I don't see why this one is any different. IMO the apiserver is easier to understand this way.

But I am very open to move parts of the implementation into pkg/audit, but leave the `WithAudit` here.